units_cpp/game_view.cc

```diff
@@ -29,7 +29,7 @@ void GameView::generate_build_units_actions(std::vector<Action> &actions,
       view_player.tech.includes(*begin->second.get_tech()) &&
       begin->second.get_cost() <= coins) {
     units.push_back(begin->first);
-    generate_build_units_actions(actions, units, coins - begin->second.get_cost(), max_units - 1, next_it, end);
+    generate_build_units_actions(actions, units, coins - begin->second.get_cost(), max_units - 1, begin, end);
```

Comment:
Consider reducing the number of arguments of this function (namely by returning pair<> and taking in the container instead of begin and end)